cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -19,16 +19,30 @@
 # instruction sets that would boost performance.
 include(CheckCXXCompilerFlag)
 # x86/amd64 compiler flags
-check_cxx_compiler_flag("-msse4.2" CXX_SUPPORTS_SSE4_2)
-check_cxx_compiler_flag("-mavx2" CXX_SUPPORTS_AVX2)
-check_cxx_compiler_flag("-mavx512f" CXX_SUPPORTS_AVX512)
+if(MSVC)
+  set(ARROW_SSE42_FLAG "")
+  set(ARROW_AVX2_FLAG "/arch:AVX2")
+  set(ARROW_AVX512_FLAG "/arch:AVX512")
+  set(CXX_SUPPORTS_SSE2 TRUE)
+  set(CXX_SUPPORTS_SSE4_2 TRUE)
+else()
+  set(ARROW_SSE2_FLAG "-msse2")
+  set(ARROW_SSE42_FLAG "-msse4.2")
```

Comment:
Could you use `ARROW_SSE4_2_FLAG` name for consistency?
Other SSE4.2 variables use `SSE4_2` instead of `SSE42`.

Reply:
Done